### PR TITLE
Use Build.ArtifactStagingDirectory now that we run with native contai…

### PR DESCRIPTION
**Pull request by ViktorHofer**
…ners in AzDO

Before when we running with a custom docker invocation we couldn't write to files outside of the mount. Now that we use the native AzDO container integration that shouldn't be an issue anymore.

### CODE REVIEW

**Review by mmitche — APPROVED** [review]

File: eng/pipelines/templates/jobs/vmr-build.yml [hunk 1]
```diff
@@ -455,6 +454,7 @@ jobs:
           customPreBuildArgs=''
           customBuildArgs=''
           extraBuildProperties=''
+          customBuildArgs="--ci --prepareMachine -c ${{ parameters.configuration }}"
```
Comment (MichaelSimons) [inline eng/pipelines/templates/jobs/vmr-build.yml:457]: How is this change related?  Also why wasn't this just done two lines above?

Comment (ViktorHofer) [reply eng/pipelines/templates/jobs/vmr-build.yml:457]: Copy / Paste error, I should have replaced L455 instead. It doesn't cause any issues but I will clean this up when I touch the file again. Thanks for pointing it out.

Unrelated change. I noticed that the Unix test invocation didn't receive these expected arguments. I didn't want to spin a new with PR just for this one liner to save resources.
